### DISCUSSION TITLE
Changing Bluetooth Name & WIFI SSID

### DISCUSSION THREAD

**Discussion by kentherzz** [ludwig-v/wireless-carplay-dongle-reverse-engineering#46] [Ideas]
Is it possible to change the name of the device using script U2W.sh? I am looking into ways to change the name of both the bluetooth interface and the Wifi SSID so I can have consistent name showing up both in iOS and when the device boot up.

I found following files:
/etc/bluetooth_name
/etc/wifi_name

I thought about writing a simple rename into the script which gets executed. There must also be some code that is detecting the type of car when you plug it in as there seem to be a name like "VOLKSW-000". Does anyone know in the code where we can find this identification - if so it should be possible to change the behaviour.

I did some experimenting with this:

```
chmod 777 /etc/bluetooth_name && sync && echo "BT: Writable"
echo "Golf GTE" > /etc/bluetooth_name && sync && echo "BT: Golf GTE"
chmod 444 /etc/bluetooth_name && sync && echo "BT: Write Protected"

chmod 777 /etc/bluetooth/hcid.conf && sync && echo "HCID: Writable"
sed -i 's/name.*/name "Golf GTE";/g' /etc/bluetooth/hcid.conf && sync && echo "HCID: Golf GTE"
chmod 444 /etc/bluetooth/hcid.conf && sync && echo "HCID: Write Protected"

chmod 777 /etc/hostapd.conf && sync && echo "SSID: Writable"
sed -i 's/ssid=.*/ssid=Golf GTE/g' /etc/hostapd.conf && sync echo && echo "SSID: Golf GTE"
chmod 444 /etc/hostapd.conf && sync && echo "SSID: Write Protected"

chmod 777 /etc/wifi_name && sync && echo "WiFi: Writable"
echo "Golf GTE" > /etc/wifi_name && sync && echo "WiFi: Golf GTE"
chmod 444 /etc/wifi_name && sync && echo "WiFi: Write Protected"
```

First I had just the 'echo' commands and no 'chmod'. This works fine when it is connected to USB at home. I can see the new SSID. As soon as I plug into my car it reverts to SSID "VOLKSWAGENXX" and BT Name " VOLKSW-XXXX". My second attempt was to try to write protect these files, and I was running the U2W.sh script at every boot. No luck. Still the result is the same. I am guessing there is something happening when you connect device to the car and there must be some code in the binary files.

As there are many people interested in GUI and modification, I hope there are some others interested in this topic as well :-)

**Comment by ghost** [comment]
Have you tried to SSH into the device after the changes and type 'sync' this will clear cache. 

You could also try and change owner of the files, but be very careful if for some reason you lock it and the OS can't read these files you may be SOL and you will lock yourself out. 

I don't know why they didn't just use 'CarPlay' it's logical and makes sense..... China always got concepts but execute them like a shopping cart with only 3 wheels.

**Reply by kentherzz** [reply]
No I haven't tried to use the SSH...yet. I just script anything I want to do and put it inside U2W.sh. When I dump the values after the dongle has been connected to car I can see that they have been reverted. My values stay as I want them as long as the dongle IS NOT connected to the car.  As soon as it detects my cars wired CarPlay interface and initialize things they are back to standard, VOLKSW-XXXX and standard SSID.

root should not be able to change the file just with chmod 444. But what I think happens is that the files I have listed above are being replaced and not modified.

Haha, so true that about China!
Thanks!

**Reply by hassankhan** [reply]
I think the binary responsible for setting the SSID and BT device names is [`/usr/sbin/set_wifi_mac`](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/blob/master/Extracted/2021-03-09.9001/usr/sbin/set_wifi_mac). Not super clued in on reverse engineering binaries though, would be nice to know how it gets the car make/manufacturer info.

EDIT: I was wrong; `set_wifi_mac` sets the WiFi's MAC address, nothing more.

**Reply by hassankhan** [reply]
I've tried to trace the init scripts starting from `init.d/` and still can't figure out who's setting the SSID.
